react/features/chat/middleware.js

```diff
@@ -318,6 +319,7 @@ function _handleReceivedMessage({ dispatch, getState },
     const hasRead = participant.local || isChatOpen;
     const timestampToDate = timestamp ? new Date(timestamp) : new Date();
     const millisecondsTimestamp = timestampToDate.getTime();
+    const shouldShowNotification = userSelectedNotifications['notify.chatMessages'] && !hasRead && !isReaction;
```

Comment:
shouldn't this also consider the notifications array? whether !notifications or notifications.includes('notify.chatMessages').

I would do a selector under settings/functions :
```
function isNotificationEnabled(notificationKey, statefull) {
     notifications = statefull[<config>].notifications;
     userSelectedNotifications = statefull[<settings>].userSelectedNotifications ;
     return userSelectedNotifications[notificationKey] && (!notifications  || notifications.includes(notificationKey));
}
```
There might already be a selector for checking whether notif is enabled based solely on the config array, so the above function would call that and also check the user selected setting from the more tab.

so it would become `const shouldShowNotification = isNotificationEnabled('notify.chatMessages') && !hasRead && !isReaction;`

Reply:
nevermind, seems the check is done on the showNotification action side :)